File: 70.climbing-stairs.go
// @leet start
func climbStairs(n int) int {
  if n == 1 { return 1 }
  a, b := 1, 2
  for i := 3; i <= n; i++ {
    a, b = b, a + b
  }
  return b
}
// @leet end

File: 70.climbing-stairs.js
// @leet start
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
  if (n === 1) return 1;
  let res = [1, 2];
  for (let i = 2; i < n; i++) {
    res[1] = res[0] + res[1];
    res[0] = res[1] - res[0];
  }
  return res[1];
};
// @leet end
